api/apikeys.go

```diff
@@ -58,3 +158,47 @@ func (api *API) userAPIKeyDELETE(u *database.User, w http.ResponseWriter, req *h
 	}
 	api.WriteSuccess(w)
 }
+
+// userAPIKeyPUT updates an API key. Only possible for public API keys.
+func (api *API) userAPIKeyPUT(u *database.User, w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	akID, err := primitive.ObjectIDFromHex(ps.ByName("id"))
+	if err != nil {
+		api.WriteError(w, err, http.StatusBadRequest)
+		return
+	}
+	var body APIKeyPUT
+	err = parseRequestBodyJSON(req.Body, LimitBodySizeLarge, &body)
+	if err != nil {
+		api.WriteError(w, err, http.StatusBadRequest)
+		return
+	}
+	err = api.staticDB.APIKeyUpdate(req.Context(), *u, akID, body.Skylinks)
+	if err != nil {
+		api.WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	api.WriteSuccess(w)
+}
+
+// userAPIKeyPATCH patches an API key. The difference between PUT and PATCH is
+// that PATCH only specifies the changes while PUT provides the expected list of
+// covered skylinks. Only possible for public API keys.
+func (api *API) userAPIKeyPATCH(u *database.User, w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	akID, err := primitive.ObjectIDFromHex(ps.ByName("id"))
```

Comment:
same answer...

it might be better to start referring to all occurrences of an issue in the original comment, rather than commenting "same" - it sometimes get confusing.